File: random contests/find and replace.cpp
#include <iostream>
    #include <string>
    using namespace std;
    int main()
    {
        int t;
        cin>>t;
        while(t--)
        {
            int n;
            cin>>n;
            string s;
            cin>>s;
            int flag=1;
            for(int i=0;i<n;i++)
            {
                for(int j=i+1;j<n;j+=2)
                {
                    if(s[i]==s[j])
                    {
                        flag=0;
                        break;
                    }
                }
            }
            if(flag)
            {
                cout<<"YES"<<endl;
            }
            else
            {
                cout<<"NO"<<endl;
            }
        }
        return 0;
    }

File: random contests/find and replace.py
for i in range(int(input())):
    x=int(input())
    s=input()
    a=1
    for i in range(x):
        for j in range(i+1,x,2):
            if s[i]==s[j]:
                a=0
                break
    if a:
        print("YES")
    else:
        print("NO")